### PR TITLE
Improvement/terraform-to-secrets

### PR DESCRIPTION
**Pull request by felddy**
# 

## 🗣 Description

- add support for creating secrets from other pieces of Terraform state.
- add support for processing multiple IAM access keys at once
- simplified options
- resources can now be tagged with `GitHub_Secret_Name` and `GitHub_Secret_Terraform_Lookup` to generate adhoc secrets.
 renamed tool to `terraform-to-secrets` since it handles more than IAM creds now.

For example a resource tagged with:
 - GitHub_Secret_Name: `BUILD_ROLE_TO_ASSUME`
 - GitHub_Secret_Terraform_Lookup: `arn`

Would create a secret `BUILD_ROLE_TO_ASSUME` with the contents of the `arn` value

## 💭 Motivation and Context

- @dav3r asked for it.
- It makes sense.

## 🧪 Testing

- This has been used in production with great success.
- Used the `--state` and `--dry-run` options to feed in artisanal json with various scenarios.

## 📷 Screenshots (if appropriate)
Screen Shot 2020-02-24 at 16 03 16

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: project_setup/scripts/terraform-to-secrets [hunk 1]
```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python
+
+"""Parse Terraform state and create GitHub secrets.
+
+Secrets are create for IAM access keys, and specially tagged resources.
```
Comment (jsf9k) [inline project_setup/scripts/terraform-to-secrets]: "create" --> "created"

File: project_setup/scripts/terraform-to-secrets [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+Resources tagged with "GitHub_Secret_Name" and "GitHub_Secret_Terraform_Lookup" will
+have a single secret created based on the tag contents.  For example if a resource had
+these tags:
+ - GitHub_Secret_Name: BUILD_ROLE_TO_ASSUME
+ - GitHub_Secret_Terraform_Lookup: arn
+
+A secret would be generated with the name "BUILD_ROLE_TO_ASSUME" and the value of
+the resource's ARN.
+
+This command must be executed in the directory containing a .terraform state
```
Comment (jsf9k) [inline project_setup/scripts/terraform-to-secrets]: Should be "executed in the directory containing the" or "executed in a directory containing a", not mixed.

**Review by jsf9k — APPROVED** [review]
I made two comments regarding two minor grammatical things, but otherwise approved.

File: project_setup/scripts/terraform-to-secrets [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+            ["terraform", "show", "--json"], stdout=subprocess.PIPE
+        )
+        data = process.stdout
+    # Normally we'd check the process return code here.  But Terraform is perfectly
+    # happy to return zero even if there were no state files.
+    json_state: Dict = json.loads(data)
+
+    if not json_state.get("values"):
+        logging.critical("Is there a .terraform state directory here?")
+        raise Exception("No terraform state found.")
```
Comment (dav3r) [inline project_setup/scripts/terraform-to-secrets]: "terraform" -> "Terraform"

File: project_setup/scripts/terraform-to-secrets [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+"""Parse Terraform state and create GitHub secrets.
+
+Secrets are create for IAM access keys, and specially tagged resources.
+
+For each IAM access key two secrets will be created:
+ - AWS_ACCESS_KEY_ID
+ - AWS_SECRET_ACCESS_KEY
+
+If there is only one access key, the secrets will be named as above.  If more than one
+key exists in the Terraform state, the key's name will be appended to the secret's name.
```
Comment (dav3r) [inline project_setup/scripts/terraform-to-secrets]: Isn't it the user's name (owner of the key) that is appended, not the key's name?

**Review by dav3r — APPROVED** [review]
Super sweet!  🍬 
I noticed a couple of other small things, along with what @jsf9k already mentioned.

File: project_setup/scripts/terraform-to-secrets (hunk 4 above)
Comment (felddy) [reply project_setup/scripts/terraform-to-secrets]: Yup.  I was lying.

**Pull request merged**